mitiq/shadows/shadows_utils.py

```diff
@@ -0,0 +1,101 @@
+# Copyright (C) Unitary Fund
+# Portions of this code have been adapted from PennyLane's tutorial
+# on Classical Shadows.
+# Original authors: PennyLane developers: Brian Doolittle, Roeland Wiersema
+# Tutorial link: https://pennylane.ai/qml/demos/tutorial_classical_shadows
+#
+# This source code is licensed under the GPL license (v3) found in the
+# LICENSE file in the root directory of this source tree.
+
+"""Defines utility functions for classical shadows protocol."""
+from typing import Tuple, List
+
+import numpy as np
+from numpy.typing import NDArray
+
+import mitiq
+
+
+def n_measurements_tomography_bound(epsilon: float, num_qubits: int) -> int:
+    """
+    This function returns the minimum number of classical shadows required
+    for state reconstruction for achieving the desired accuracy.
+
+    Args:
+        epsilon: The error on the estimator.
+        num_qubits: The number of qubits in the system.
+
+    Returns:
+        An integer that gives the number of snapshots required to satisfy the
+        shadow bound.
+    """
+    return int(34 * (4**num_qubits) * epsilon ** (-2))
+
+
+def local_clifford_shadow_norm(obs: mitiq.PauliString) -> float:
+    """
+    Calculate shadow norm of an operator with random unitary sampled from local
+    Clifford group.
+    Args:
+        opt: a self-adjoint operator.
```

Comment:
Needs updating now that this function takes a mitiq `PauliString`.

Reply:
Completed in [a6c5793](https://github.com/unitaryfund/mitiq/pull/1907/commits/a6c579348fe6c55b0291dce80a3012d0106c64a2).